RFC: allow string interpolation in L"..."

simeonschaub opened the pull request:
This uses `%$` for interpolation, as proposed by you in #27, which seems fine to me, but I'm happy to change it for something else. Still need to add docs and tests, but wanted to get a go-ahead on this approach first.
fixes #27

simeonschaub commented on src/LaTeXStrings.jl:
```diff
@@ -29,21 +29,54 @@ struct LaTeXString <: AbstractString
+    parseatom(s, i; filename=nothing) = Meta.parse(s, i, greedy=false)
+end
+
+macro L_str(s, flags...)
```
Is there a reason this accepts a flag as well? Perhaps we could add an `r` flag, that skips interpolation.

stevengj replied:
We can drop the flags since we don't use them.

simeonschaub commented on src/LaTeXStrings.jl:
```diff
@@ -29,21 +29,54 @@ struct LaTeXString <: AbstractString
+    position(buf) > 0 && push!(ex.args, String(take!(buf)))
+    return :(LaTeXString($(esc(ex))))
+end
 macro L_mstr(s, flags...) latexstring(s) end
```
Was this added on purpose?

stevengj replied:
It used to be (before Julia 1.0) that you needed separate `_str` and `_mstr` macros for multi- and single-line strings.   Can be deleted now.

stevengj commented on src/LaTeXStrings.jl:
```diff
@@ -29,21 +29,54 @@ struct LaTeXString <: AbstractString
+                atom !== nothing && push!(ex.args, atom)
+                continue
+            else
+                print(buf, '%')
```
Isn't this redundant with the `print(buf, c)` below?

simeonschaub replied:
It shouldn't be, since `c` is already the character after `%`, so we first print `%` and then print `c` outside the `if` statement below.

simeonschaub replied:
There was actually still a bug in the way I handled this, because `%%$(...)` wasn't interpolated correctly. Should be fixed now.

stevengj commented on src/LaTeXStrings.jl:
```diff
@@ -29,21 +29,54 @@ struct LaTeXString <: AbstractString
+        c = @inbounds s[i]
+        if c === '%'
+            i = nextind(s, i)
+            i > ncodeunits(s) && break
```
Does this mean that if a `%` is the last character of the string then it will be lost?

simeonschaub replied:
Yes, good catch! Will fix that!

simeonschaub commented:
Ok, I refactored some of the logic a bit and fixed some bugs I discovered, but I think I fixed all of them now. I hope it's okay, I added some doctests and included them in the tests, but I was afraid that if we didn't test those, they would bitrot. I also bumped the minor version, but feel free to revert that if you want to do something else for releases.

stevengj commented:
LGTM.
